homeassistant/components/zabbix/__init__.py

```diff
@@ -68,101 +97,292 @@
+    strings = {}
+    try:
+        _state_as_value = float(state.state)
+        floats[entity_id] = _state_as_value
+    except ValueError:
+        try:
+            _state_as_value = float(state_helper.state_as_number(state))
+            floats[entity_id] = _state_as_value
+        except ValueError:
+            strings[entity_id] = state.state
+    for key, value in state.attributes.items():
+        # For each value we try to cast it as float
+        # But if we cannot do it we store the value
+        # as string
+        attribute_id = f"{entity_id}/{key}"
+        try:
+            float_value = float(value)
+        except (ValueError, TypeError):
+            float_value = None
+        if float_value is None or not math.isfinite(float_value):
+            strings[attribute_id] = str(value)
+        else:
+            floats[attribute_id] = float_value
+    float_keys_count = len(float_keys)
+    float_keys.update(floats.keys())
+    if len(float_keys) != float_keys_count:
+        floats_discovery = [
+            {"{#KEY}": str(float_key)[:230]} for float_key in float_keys
+        ]
+        metric = ItemValue(
+            publish_states_host,
+            "homeassistant.floats_discovery",
+            json.dumps(floats_discovery),
+        )
+        metrics.append(metric)
+    for key, value in floats.items():
+        metric = ItemValue(
+            publish_states_host, f"homeassistant.float[{str(key)[:230]}]", value
+        )
+        metrics.append(metric)
 
-    conf = config[DOMAIN]
-    protocol = "https" if conf[CONF_SSL] else "http"
+    string_keys.update(strings)
 
-    url = urljoin(f"{protocol}://{conf[CONF_HOST]}", conf[CONF_PATH])
-    username = conf.get(CONF_USERNAME)
-    password = conf.get(CONF_PASSWORD)
+    return metrics
 
-    publish_states_host = conf.get(CONF_PUBLISH_STATES_HOST)
 
-    entities_filter = convert_include_exclude_filter(conf)
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Zabbix component from a ConfigEntry."""
+    zabbix_sender: AsyncSender
+    host: str
+    filter_dict: dict[str, list[str]] = {}
+    hass_data: dict
+    instance: ZabbixThread
 
-    try:
-        zapi = ZabbixAPI(url=url, user=username, password=password)
-        _LOGGER.info("Connected to Zabbix API Version %s", zapi.api_version())
-    except ZabbixAPIException as login_exception:
-        _LOGGER.error("Unable to login to the Zabbix API: %s", login_exception)
-        return False
-    except HTTPError as http_error:
-        _LOGGER.error("HTTPError when connecting to Zabbix API: %s", http_error)
-        zapi = None
-        _LOGGER.error(RETRY_MESSAGE, http_error)
-        event_helper.call_later(
-            hass,
-            RETRY_INTERVAL,
-            lambda _: setup(hass, config),  # type: ignore[arg-type,return-value]
-        )
+    hass.data.setdefault(DOMAIN, {})
+    # Already configured from configuration.yaml. Not allow UI config.
```

Comment:
Why do we still allow YAML configuration? Is there a reason we still maintain the YAML configuration? When adding a config flow we actually want to get every user over to UI, so what can be done to allow that?

Reply:
It is for backward compatability for the users having zabbix integration configured with YAML (old style). 
The reason - after new HA upgrade exising zabbix integration will not broke for the ones who uses it. Later all them can switch to use the configuration with the UI, but it will be their choice of time when to do so.
New users will use the config UI flow anyway, as it is much simplier, intuitive and provides more options.

Reply:
When implementing we want to migrate all users over. I think we have to find a sweet spot for zabbix, because not all YAML has to be turned into a config flow because Zabbix is a service for exposing data to other services. Azure Data Explorer got added recently and also has some YAML to configure that part, but all the credentials and connection related info is stored in the config flow